### PR TITLE
[Feature/extensions] Migrates Get Detector action to extension

### PR DESCRIPTION
**Pull request by owaiskazi19**
### Description

Migrates the GetAnomalyDetectorTransportAction, their Rest Handlers, and other supporting classes from plugin to extension.

This PR also brings back the original index name for `ANOMALY_DETECTORS_INDEX` from
`opendistro-anomaly-detectors `->  `.opendistro-anomaly-detectors`

Removed unused files: `RestGetDetectorAction`, `RestCreateDetectorAction` and `RestValidateDetectorAction`

### Issues Resolved
Fixes https://github.com/opensearch-project/opensearch-sdk-java/issues/373

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/main/java/org/opensearch/ad/transport/GetAnomalyDetectorTransportAction.java [hunk 1]
```diff
@@ -1,430 +1,438 @@
...
+        List<EntityProfileName> defaultEntityProfiles = Arrays.asList(EntityProfileName.STATE);
+        this.defaultEntityProfileTypes = new HashSet<EntityProfileName>(defaultEntityProfiles);
+
+        this.xContentRegistry = xContentRegistry;
+        this.nodeFilter = nodeFilter;
+        filterByEnabled = AnomalyDetectorSettings.FILTER_BY_BACKEND_ROLES.get(settings);
+        try {
+            clusterService.getClusterSettings().addSettingsUpdateConsumer(FILTER_BY_BACKEND_ROLES, it -> filterByEnabled = it);
+        } catch (Exception e) {
+            // FIXME Handle this
```
Comment (joshpalis) [inline src/main/java/org/opensearch/ad/transport/GetAnomalyDetectorTransportAction.java:128]: https://github.com/opensearch-project/opensearch-sdk-java/issues/422 has been closed, no need to surround the `addSettingsUpdateConsumer `with a try/catch

File: src/main/java/org/opensearch/ad/transport/GetAnomalyDetectorTransportAction.java [hunk 2]
```diff
@@ -1,430 +1,438 @@
...
+        }
+
+        this.transportService = transportService;
+        this.adTaskManager = adTaskManager;
+    }
+
+    public void doExecute(Task task, GetAnomalyDetectorRequest request, ActionListener<GetAnomalyDetectorResponse> actionListener) {
+        String detectorID = request.getDetectorID();
+        // Temporary null user for AD extension without security. Will always execute detector.
+        UserIdentity user = getNullUser();
```
Comment (joshpalis) [inline src/main/java/org/opensearch/ad/transport/GetAnomalyDetectorTransportAction.java:139]: interesting, Ill make sure to bypass user resolution this way for `start detector`

**Review by joshpalis — COMMENTED** [review]
Doing an initial pass

File: src/main/java/org/opensearch/ad/transport/GetAnomalyDetectorTransportAction.java (hunk 1 above)
Comment (owaiskazi19) [reply src/main/java/org/opensearch/ad/transport/GetAnomalyDetectorTransportAction.java]: Thanks for pointing to the issue. I went ahead and removed the FIXME (try/catch) for all the other files as well.

File: src/main/java/org/opensearch/ad/model/AnomalyDetector.java [hunk 3]
```diff
@@ -72,7 +72,7 @@ public class AnomalyDetector implements Writeable, ToXContentObject {
         it -> parse(it)
     );
     public static final String NO_ID = "";
-    public static final String ANOMALY_DETECTORS_INDEX = "opendistro-anomaly-detectors";
+    public static final String ANOMALY_DETECTORS_INDEX = ".opendistro-anomaly-detectors";
```
Comment (joshpalis) [inline src/main/java/org/opensearch/ad/model/AnomalyDetector.java:75]: Just curious, why was the period removed for this system index name?

Comment (owaiskazi19) [reply src/main/java/org/opensearch/ad/model/AnomalyDetector.java:75]: Good question. Here is the [ref](https://github.com/opensearch-project/opensearch-sdk-java/issues/241#issuecomment-1310693864)

File: src/main/java/org/opensearch/ad/rest/RestGetAnomalyDetectorAction.java [hunk 4]
```diff
@@ -76,27 +126,52 @@ protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient cli
             buildEntity(request, detectorId)
         );
 
-        return channel -> client
-            .execute(GetAnomalyDetectorAction.INSTANCE, getAnomalyDetectorRequest, new RestToXContentListener<>(channel));
+        GetAnomalyDetectorTransportAction getTransportAction = new GetAnomalyDetectorTransportAction(
+            transportService,
+            null, // nodeFilter
+            null, // ActionFilters actionFilters
```
[the remark this answers is not in the document]
Comment (owaiskazi19) [reply src/main/java/org/opensearch/ad/rest/RestGetAnomalyDetectorAction.java:132]: That's correct

**Review by joshpalis — APPROVED** [review]

**Review by dbwiddis — APPROVED** [review]
LGTM but for ease of review would really like separate commits when uncommenting the entire file and then making changes to it.  I had to do the diff manually.

**Pull request merged**